src/components/Questions.js

```diff
@@ -0,0 +1,139 @@
+import React, { Component } from 'react';
+import { Redirect } from 'react-router-dom';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import {
+  resetTimer,
+} from '../redux/actions';
+
+function shuffleArray(received) {
+  const array = [...received];
+  for (let i = array.length - 1; i > 0; i -= 1) {
+    const j = Math.floor(Math.random() * (i + 1));
+    [array[i], array[j]] = [array[j], array[i]];
+  }
+  return array;
+}
+
+function displayButtonNext(goToNextQuestion, resetTimerGlobal) {
+  return (
+    <input
+      type="button"
+      data-testid="btn-next"
+      className="next"
+      style={{ display: 'none' }}
+      onClick={() => {
+        goToNextQuestion();
+        resetTimerGlobal();
+      }}
+      value="Próximo"
+    />
+  );
+}
+
+function colorAnswers() {
+  const wrong = Array.from(document.querySelectorAll('.wrong'));
+  const correct = document.querySelector('.correct');
+  wrong.forEach((item) => {
+    item.classList.add('wrongStyle');
+  });
+  correct.classList.add('correctStyle');
+  const next = document.querySelector('.next');
+  next.style.display = 'block';
+}
+
+function renderCorrectInput(elem, addCorrectAssertion) {
+  return (
+    <input
+      type="button"
+      key={elem}
+      className="correct"
+      data-testid="correct-answer"
+      value={elem}
+      onClick={() => {
+        colorAnswers();
+        addCorrectAssertion();
+      }}
+    />
+  );
+}
+
+function renderWrongInput(elem, incorrectAnswers) {
+  return (
+    <input
+      type="button"
+      key={elem}
+      className="wrong"
+      data-testid={`wrong-answer-${incorrectAnswers.indexOf(elem)}`}
+      value={elem}
+      onClick={() => colorAnswers()}
+    />
+  );
+}
```

Comment:
Mano, nao sei direito como ta o estado do redux mas se nao tiver, acho interessante criar um estado (vinculado a action onClick no botao das alternativas) que vai informar se a pergunta foi respondida. Tipo um `isAnswered: true/false`. Dessa forma vc nao iria precisar dessa função colorAnswers e poderia fazer assim pras duas  funções de input que vc criou:
```suggestion
function renderCorrectInput(elem, addCorrectAssertion) {
  return (
    <input
      type="button"
      key={elem}
      className={isAnswered ? 'green-border' : ''}
      disabled={isAnswered}
      data-testid="correct-answer"
      value={elem}
      onClick={() => {
        colorAnswers();
        addCorrectAssertion();
      }}
    />
  );
}

function renderWrongInput(elem, incorrectAnswers) {
  return (
    <input
      type="button"
      key={elem}
      className={isAnswered ? 'red-border' : ''}
      disabled={isAnswered}
      data-testid={`wrong-answer-${incorrectAnswers.indexOf(elem)}`}
      value={elem}
      onClick={() => colorAnswers()}
    />
  );
}
```

Reply:
Top demais Ale!